mlir/include/mlir/ExecutionEngine/ExecutionEngine.h

```diff
@@ -99,6 +99,9 @@ struct ExecutionEngineOptions {
   /// If `enablePerfNotificationListener` is set, the JIT compiler will notify
   /// the llvm's global Perf notification listener.
   bool enablePerfNotificationListener = true;
+
+  /// Setting initialize=false to defer initialization
+  bool shouldInitialize = true;
```

Comment:
I'm not sure why we need a flag for this, seems to me that since we introduce a separate `initialize()` method we don't need an option and users can just call `initialize()` separately.

Reply:
> I'm not sure why we need a flag for this, seems to me that since we introduce a separate `initialize()` method we don't need an option and users can just call `initialize()` separately.

It is introduced to reduce the changes downstream has to make.

Reply:
That's not the usual criteria of development in LLVM though: we aim to keep the code here maintainable and preserve our ability to innovate. Downstream know they need to adapt when we find design issues we fix upstream.